### ISSUE TITLE
[Feat]: appium-doctor add driver-level options.

### ISSUE THREAD

**Issue by youngfreeFJS** [appium/appium#18302]
**Is your feature request related to a problem? Please describe.**
When we use `Mac2 driver`, we want to diagnose and fix `Mac2 driver` configuration.
But now only support device-level options, like `appium-doctor --ios` or `appium-doctor --android`
If I want to contribute to appium and add driver-level options to the doctor (like: `appium-doctor --mac2`), does it meet the design expectations?

**Describe the solution you'd like**
add option: `appium-doctor --driver  ${diverName}`, like: `appium-doctor --driver mac2` or `appium-doctor --driver expresso`

**Describe alternatives you've considered**
add option: `appium-doctor --${diverName}`, like `appium-doctor --mac` or  `appium-doctor --expresso`

**Comment by jlipps** [comment]
Yes, this is the plan. Drivers will export their own doctor extensions which will be run when you run `appium doctor`. That way Appium Doctor itself can be a minimal check/reporting library and the actual validations will live next to related code, that is in the driver/plugins themselves.

**Comment by youngfreeFJS** [comment]
> Yes, this is the plan. Drivers will export their own doctor extensions which will be run when you run `appium doctor`. That way Appium Doctor itself can be a minimal check/reporting library and the actual validations will live next to related code, that is in the driver/plugins themselves.

In more detail, I think the appium@doctor package can provide an execution mechanism for minimal register and running driver doctor checks. Extensions/ plugins can write a check script in themselves, appium@doctor find each script by  APPIUM_HOME absolute path,  configuration file or something else.

And the appium@doctor package can provide some public check function as a `check marketplace`, contain like Node.js check,  Xcode check, Android developer tools check, etc.

---- 

What I'm not sure is whether the extension/plugin should provide an public protocol to indicate what capabilities themselves can provide. 
Example like this issue: [[Feat]: provide default caps file in Capability Sets.](https://github.com/appium/appium-inspector/issues/732.) driver can provide default caps file or something of the agreement specifies what it implements.

In this way, each extension/plugin can achieve corresponding functions in a more standardized manner, while being decoupled from the appium itself.

**Any feedback would be appreciated.** 😊 @jlipps

**Comment by eglitise** [comment]
Driver-level options have been added starting from:
- `appium` `2.4.1`
- `appium-uiautomator2-driver` `2.39.0`
- `appium-espresso-driver` `2.31.0`
- `appium-xcuitest-driver` `5.13.0`
- `appium-mac2-driver` `1.9.0`

See [documentation](https://appium.io/docs/en/latest/cli/extensions/#doctor) for how these can be executed.
